crates/turborepo-scm/src/git.rs

```diff
@@ -244,19 +348,28 @@ mod tests {
     use std::{
         assert_matches::assert_matches,
         collections::HashSet,
-        fs,
+        env, fs,
         path::{Path, PathBuf},
         process::Command,
+        sync::{Mutex, OnceLock},
     };
 
     use git2::{Oid, Repository, RepositoryInitOptions};
-    use tempfile::TempDir;
+    use tempfile::{NamedTempFile, TempDir};
     use test_case::test_case;
     use turbopath::{AbsoluteSystemPath, AbsoluteSystemPathBuf, PathError};
     use which::which;
 
     use super::{previous_content, ChangedFiles};
-    use crate::{Error, Git, SCM};
+    use crate::{
+        git::{GitHubCommit, GitHubEvent},
+        Error, Git, SCM,
+    };
+
+    /// By default `#[test_case]` will run in parallel.
+    /// This is a problem for testing things like environment variables, which
+    /// are not thread-safe. This lock ensures that these tests run serially.
+    static TEST_PARALLELISM_LOCK: OnceLock<Mutex<bool>> = OnceLock::new();
```

Comment:
This isn't a bad solution per se, but just noting that when we had this bug previously,  we created `TestCase` structs that we [manually looped over](https://github.com/vercel/turborepo/blob/2e3f7e730a921666b49a5d561b702a8353b2f86e/crates/turborepo-ci/src/lib.rs#L125).

Reply:
this ended up getting removed in the midst of hoisting the env beyond the functions in question.